aptos-move/aptos-vm/src/move_vm_ext/vm.rs

```diff
@@ -44,6 +44,10 @@ impl MoveVmExt {
         extensions.add(NativeTransactionContext::new(script_hash));
         extensions.add(NativeCodeContext::default());
 
+        // The VM code loader has bugs around module upgrade. After a module upgrade, the internal
+        // cache needs to be flushed to work around those bugs.
+        self.inner.flush_loader_cache_if_invalidated();
```

Comment:
Why is this the right place to flush?

Reply:
Forgot to answer: this is a good place because we use one session per transaction execution. So you ensure that for the next session, the cache is clear.